beater/connbeat.go

```diff
@@ -50,32 +55,45 @@ func (cb *Connbeat) exportConnection(c Connection) error {
 		"remote_ip":     c.remoteIp,
 		"remote_port":   c.remotePort,
 		"local_process": c.process,
+		"beat": common.MapStr{
+			"local_ips": localIps.ToSlice(),
+		},
 	}
 
 	cb.events.PublishEvent(event)
 
 	return nil
 }
 
-func (cb *Connbeat) Run(b *beat.Beat) error {
+func (cb *Connbeat) Pipe(connectionListener <-chan Connection, serverConnectionListener <-chan ServerConnection) error {
 	var err error
 
-	connectionListener, serverConnectionListener := Listen()
+	localIps := mapset.NewSet()
 
 	for {
 		select {
 		case <-cb.done:
 			return nil
 		case c := <-connectionListener:
-			err = cb.exportConnection(c)
+			localIps.Add(c.localIp)
+			err = cb.exportConnection(c, localIps)
 		case s := <-serverConnectionListener:
-			err = cb.exportServerConnection(s)
+			if s.localIp != "0.0.0.0" {
+				localIps.Add(s.localIp)
```

Comment:
Correct me if I'm wrong, but this gives us only ip's a server is listening on (which would be none if all were listening on 0.0.0.0) and ip's over which client connections are coming in/going out (which will typically be sufficient)? Moreover initially we probably won't get all local ips reported in the JSON output.

As far as I can tell both will not really be a problem, the former will be fine since the ip's that have connections are the only ip's we are interested in anyway and the latter will be fixed by the regular updates of the same process. It would be a good idea though to mention this in the README where you describe the data format though.